mailbox/store/src/test/java/org/apache/james/mailbox/store/AbstractCombinationManagerTest.java

```diff
@@ -123,162 +129,187 @@ public void searchFromMailboxManagerShouldReturnMessagesUsingSetInMailboxesFromM
     }
 
     @Test
     public void getMessageFromMessageIdManagerShouldReturnMessageWhenMoveMessageWithMailboxIdFromMailboxManager() throws Exception {
-        MessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
 
-        mailboxManager.moveMessages(MessageRange.all(), MailboxManagerFixture.MAILBOX_PATH2, MailboxManagerFixture.MAILBOX_PATH1, session);
+        mailboxManager.moveMessages(MessageRange.all(), MailboxManagerFixture.MAILBOX_PATH1, MailboxManagerFixture.MAILBOX_PATH2, session);
 
-        assertThat(messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session)).hasSize(1);
+        List<MessageResult> listMessages = messageIdManager.getMessages(ImmutableList.of(messageId), FetchGroupImpl.MINIMAL, session);
+
+        assertThat(listMessages).hasSize(1);
+        assertThat(listMessages).extractingResultOf("getMailboxId").containsOnly(mailbox2.getMailboxId());
     }
 
     @Test
     public void getMessagesFromMessageManagerShouldReturnMessagesCreatedBySetInMailboxesFromMessageIdManager() throws Exception {
-        MessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
+        MessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, FLAGS).getMessageId();
 
         messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
 
-        assertThat(messageManager.getMessages(MessageRange.all(), FetchGroupImpl.MINIMAL, session)).hasSize(1);
+        assertThat(messageManager2.getMessages(MessageRange.all(), FetchGroupImpl.MINIMAL, session)).hasSize(1);
     }
 
     @Test
     public void getMetadataFromMessageManagerShouldReturnRecentMessageWhenSetInMailboxesFromMessageIdManager() throws Exception {
         Flags recent = new Flags(Flag.RECENT);
-        ComposedMessageId messageId = messageManager.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
+        ComposedMessageId messageId = messageManager1.appendMessage(new ByteArrayInputStream(SUBJECT_TEST_MAIL), new Date(), session, false, recent);
 
         messageIdManager.setInMailboxes(messageId.getMessageId(), ImmutableList.of(mailbox1.getMailboxId(), mailbox2.getMailboxId()), session);
 
-        messageManager = testingData.createMessageManager(mailbox2, session);
-        assertThat(messageManager.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getRecent()).containsOnly(messageId.getUid());
+        assertThat(messageManager2.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getRecent()).containsOnly(messageId.getUid());
```

Comment:
Nope, we don't know that.

messageId.getUid() is a mailbox 1 specific value. We are just lucky the UID is the same in mailbox 2.

You should before the setInMailboxes : 

```
MessageUid mailbox2NextUid = messageManager2.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getNextUid();
```

Then after the setInMailboxes

```
assertThat(messageManager2.getMetaData(true, session, FetchGroup.UNSEEN_COUNT).getRecent().asLong())
    .isGreaterThanOrEqualTo(mailbox2NextUid.asLong());
```